[Scala 3] Add tests for formatting trait parameters, local type declaration and new type wildcard symbol

tgodzik opened the pull request:
~~Scala 3 added support for specifying trait constructor parameters, which seems to work out of the box. I added some tests just in case.~~

~~EDIT: Seems that it's also working fine in case of local type declarations, that can now be defined inside a local block. Added tests for that also.~~ 

It seems that a number of new features work out of the box in scalafmt so I added tests that check:
- local type declaration
- new wildcard symbol `?`
- trait parameters

I decided to put everything into single PR instead of each in a separate one, because it seems all those are small features/changes.

kitbellew commented on scalafmt-tests/src/test/resources/dotty/Type.stat:
```diff
@@ -0,0 +1,44 @@
+      ]
+  ]
+  val a: T
+}
```
missing LF

tgodzik replied:
Fixed!

olafurpg reviewed (approved):
LGTM 👍 This is safe to merge IMO given it has no code changes

kitbellew reviewed (changes requested):
I looked over the codebase looking to make sure that (`Token.KwTrait`, `Token.KwClass`) and (`Defn.Class`, `Defn.Trait`) are always used together.

There is at least example when this is not true: `FormatOps.verticalMultiline`.

tgodzik commented:
> I looked over the codebase looking to make sure that (`Token.KwTrait`, `Token.KwClass`) and (`Defn.Class`, `Defn.Trait`) are always used together.
> 
> There is at least example when this is not true: `FormatOps.verticalMultiline`.

I added it there and added some more tests for verticalMultiline. I haven't really been able to find a test case that would fail without the change though.
